app/controllers/docs/docs_controller.rb

```diff
@@ -51,9 +51,19 @@ def get_topic(topic, current_user)
       topic_view = TopicView.new(topic.id, current_user)
       guardian = Guardian.new(current_user)
 
+      ip = request.remote_ip
+      user_id = (current_user.id if current_user)
+      track_visit = should_track_visit_to_topic?
+
+      TopicsController.defer_track_visit(topic.id, ip, user_id, track_visit)
+
       TopicViewSerializer.new(topic_view, scope: guardian, root: false)
     end
 
+    def should_track_visit_to_topic?
+      !!((!request.format.json? || params[:track_visit]) && current_user)
```

Comment:
no ... not sure about this... anon can still visit the KB and make JSON requests. Is the current_user check needed?

Reply:
yes, that is only used to determine if TopicUser should be created

https://github.com/discourse/discourse/blob/main/app/controllers/topics_controller.rb#L1176